### PR TITLE
Fix sync early exit bug. Track new session during sync. Simplify thre…

### PR DESCRIPTION
**Pull request by plorenz**
…ading and tighten up locking. Add dump api-sessions debug operation

### CODE REVIEW

File: controller/server/controller.go
Comment (andrewpmartinez) [inline controller/server/controller.go:266]: Commented out?

File: controller/sync_strats/sync_instant.go
Comment (andrewpmartinez) [inline controller/sync_strats/sync_instant.go:449]: Were these meant to be `Info`?

File: controller/server/controller.go
Comment (plorenz) [reply controller/server/controller.go:266]: For testing purposes, need to keep the massive set of expired api tokens around. Will put back in before merging. Please don't approve until it's back in ;)

File: router/handler_edge_ctrl/apiSessionAdded.go
Comment (andrewpmartinez) [inline router/handler_edge_ctrl/apiSessionAdded.go:92]: I assume the assumption is that if this sync fails, the ApiSession created during the sync will be present in the subsequent sync.

Comment (plorenz) [reply router/handler_edge_ctrl/apiSessionAdded.go:92]: Yes

**Review by andrewpmartinez — APPROVED** [review]

File: controller/server/controller.go
Comment (andrewpmartinez) [reply controller/server/controller.go:266]: Too late, approved =(

Comment (plorenz) [reply controller/server/controller.go:266]: Nooooooooo!!

Comment (plorenz) [reply controller/server/controller.go:266]: We need some marker file that will fail the build with reminders of things to do before merge

File: controller/env/broker.go
Comment (andrewpmartinez) [inline controller/env/broker.go:185]: "which ..." ? Not critical but it leaves me on the edge of my seat.

**Review by andrewpmartinez — APPROVED** [review]

File: controller/env/broker.go
Comment (plorenz) [reply controller/env/broker.go]: Tune in for the thrilling conclusion in next week's commit :smile: 

Fixed

**Pull request merged**
